projects/packages/publicize/src/rest-api/class-connections-controller.php

```diff
@@ -0,0 +1,434 @@
+		}
+
+		return $test_results_map;
+	}
+
+	/**
+	 * Get the connections data, either directly or by calling the WPCOM API
+	 *
+	 * @param bool $run_tests Whether to run the tests to check the health of the connections.
+	 *
+	 * @return array Connection objects in the same shape as from `get_connections`
+	 */
+	public static function get_connections_data( $run_tests = false ) {
+		// TODO decide on caching
+		return ( defined( 'IS_WPCOM' ) && IS_WPCOM ) ? self::get_connections( $run_tests ) : self::get_connections_from_wpcom( $run_tests );
+	}
+
+	/**
+	 * Proxy the request to the WPCOM API
+	 *
+	 * @param bool $run_tests Whether to run the tests to check the health of the connections.
+	 *
+	 * @return array Connection objects in the same shape as from `get_connections`
+	 */
+	protected static function get_connections_from_wpcom( $run_tests = false ) {
+		$site_id = Manager::get_site_id( true );
+		if ( ! $site_id ) {
+			return array();
+		}
+
+		$path = add_query_arg(
+			array( 'test_connections' => $run_tests ),
+			sprintf( '/sites/%d/publicize/connections', $site_id )
+		);
+
+		$response = Client::wpcom_json_api_request_as_user( sprintf( $path, $site_id ), 'v2', array( 'method' => 'GET' ) );
+
+		if ( is_wp_error( $response ) || 200 !== wp_remote_retrieve_response_code( $response ) ) {
+			// TODO log error.
+			return array();
+		}
+
+		$body = wp_remote_retrieve_body( $response );
+
+		$items = json_decode( $body, true );
+		return $items ? $items : array();
+	}
+
+	/**
+	 * Get list of connected Publicize connections.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 *
+	 * @return WP_REST_Response suitable for 1-page collection
+	 */
+	public function get_items( $request ) {
+		$items = array();
+
+		$connections = self::get_connections_data( $request->get_param( 'test_connections' ) );
+
+		foreach ( $connections as $item ) {
+			$items[] = $this->prepare_item_for_response( $item, $request );
+		}
+
+		$response = rest_ensure_response( $items );
+		$response->header( 'X-WP-Total', (string) count( $items ) );
+		$response->header( 'X-WP-TotalPages', '1' );
+
+		return $response;
+	}
+
+	/**
+	 * Filters out data based on ?_fields= request parameter
+	 *
+	 * @param array           $connection Array of info about a specific Publicize connection.
+	 * @param WP_REST_Request $request    Full details about the request.
+	 *
+	 * @return array filtered $connection
+	 */
+	public function prepare_item_for_response( $connection, $request ) {
+		$fields = $this->get_fields_for_response( $request );
+
+		$response_data = array();
+		foreach ( $connection as $field => $value ) {
+			if ( rest_is_field_included( $field, $fields ) ) {
+				$response_data[ $field ] = $value;
+			}
+		}
+
+		return rest_ensure_response( $response_data );
+	}
+
+	/**
+	 * Verify that user can access Publicize data
+	 *
+	 * @return true|WP_Error
+	 */
+	public function get_items_permission_check() {
```

Comment:
This callback in the future can be moved to a base class to share between different publicize controllers.

Reply:
Yes, I was trying to avoid introducing a base class at this stage. I was also trying to make the change a move of the endpoint, but we end up having to make so many other changes to the class, it ends up getting treated as a new file